cmd/notation/inspect.go

```diff
@@ -0,0 +1,327 @@
+			if err != nil {
+				logSkippedSignature(sigDesc, err)
+				skippedSignatures = true
+				continue
+			}
+
+			sig := signatureOutput{
+				MediaType:             sigDesc.MediaType,
+				Digest:                sigDesc.Digest.String(),
+				SignatureAlgorithm:    string(signatureAlgorithm),
+				SignedAttributes:      getSignedAttributes(envelopeContent),
+				UserDefinedAttributes: signedArtifactDesc.Annotations,
+				UnsignedAttributes:    getUnsignedAttributes(envelopeContent),
+				Certificates:          getCertificates(envelopeContent),
+				SignedArtifact: artifact{
+					MediaType: signedArtifactDesc.MediaType,
+					Digest:    signedArtifactDesc.Digest.String(),
+					Size:      signedArtifactDesc.Size,
+				},
+			}
+
+			output.Signatures = append(output.Signatures, sig)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	err = printOutput(opts.outputFormat, output)
+	if err != nil {
+		return err
+	}
+
+	if skippedSignatures {
+		return errors.New("at least one signature was skipped and not displayed")
+	}
+
+	return nil
+}
+
+func logSkippedSignature(sigDesc ocispec.Descriptor, err error) {
+	fmt.Fprintf(os.Stderr, "Warning: Skipping signature %s because of error: %v\n", sigDesc.Digest.String(), err)
+}
+
+func getSignedAttributes(envContent *signature.EnvelopeContent) map[string]string {
+	signedAttributes := map[string]string{}
+
+	signedAttributes["signingScheme"] = string(envContent.SignerInfo.SignedAttributes.SigningScheme)
+	signedAttributes["signingTime"] = envContent.SignerInfo.SignedAttributes.SigningTime.String()
+	signedAttributes["expiry"] = envContent.SignerInfo.SignedAttributes.Expiry.String()
+
+	for _, attribute := range envContent.SignerInfo.SignedAttributes.ExtendedAttributes {
+		if key, ok := attribute.Key.(string); ok {
+			if value, ok := attribute.Value.(string); ok {
+				signedAttributes[key] = value
+			}
+		}
+	}
+
+	return signedAttributes
+}
+
+func getUnsignedAttributes(envContent *signature.EnvelopeContent) map[string]string {
+	unsignedAttributes := map[string]string{}
+
+	if envContent.SignerInfo.UnsignedAttributes.TimestampSignature != nil {
+		unsignedAttributes["timestampSignature"] = string(envContent.SignerInfo.UnsignedAttributes.TimestampSignature)
+	}
+
+	if envContent.SignerInfo.UnsignedAttributes.SigningAgent != "" {
+		unsignedAttributes["signingAgent"] = envContent.SignerInfo.UnsignedAttributes.SigningAgent
+	}
+
+	return unsignedAttributes
+}
+
+func getCertificates(envContent *signature.EnvelopeContent) []certificateOutput {
+	certificates := []certificateOutput{}
+
+	for _, cert := range envContent.SignerInfo.CertificateChain {
+		h := sha1.Sum(cert.Raw)
+		fingerprint := strings.ToLower(hex.EncodeToString(h[:]))
+
+		certificate := certificateOutput{
+			SHA1Fingerprint: fingerprint,
+			IssuedTo:        cert.Subject.String(),
+			IssuedBy:        cert.Issuer.String(),
+			Expiry:          cert.NotAfter.String(),
```

Comment:
According to spec, those datetime should be in the RFC3339 form for the JSON output

Reply:
Please note that the current format does not match the format for the tree view as in the spec.